homeassistant/components/recorder.py

```diff
@@ -302,8 +303,7 @@ def block_till_done(self):
 
     def _setup_connection(self):
         """Ensure database is ready to fly."""
-        db_path = self.hass.config.path(DB_FILE)
```

Comment:
I had to move the config.path() call to the class initializer because _setup_connection runs in a separate thread that wasn't picking up the mocked version even when I had self.hass.start() and self.wait_recording_done() in the patch context.